samples/subsys/tracing/prj.conf

```diff
@@ -5,3 +5,4 @@ CONFIG_MP_MAX_NUM_CPUS=1
 # Debugging
 CONFIG_DEBUG_OPTIMIZATIONS=n
 CONFIG_DEBUG_THREAD_INFO=y
+CONFIG_GPIO=y
```

Comment:
drop -- this is only for GPIO tracing AFAICT and enabled in prj_gpio.conf (https://docs.zephyrproject.org/latest/samples/subsys/tracing/README.html#usage-for-gpio-tracing-backend).

Can you maybe double check whatever grep logic you used to ensure it's only for sample actually using gpio API, and in their main.c, that you enable the Kconfig? I suspect this particular sample might not be the only one where this happened.